vllm/model_executor/guided_decoding/xgrammar_decoding.py

```diff
@@ -267,6 +305,9 @@ def _ensure_ctx(self):
                 self.ctx = compiler.compile_grammar(self.config.grammar_str)
             elif self.config.json_object:
                 self.ctx = compiler.compile_builtin_json_grammar()
+            elif self.config.regex_str:
+                regex_grammar = xgr.Grammar.from_regex(self.config.regex_str)
+                self.ctx = compiler.compile_grammar(regex_grammar)
```

Comment:
Just using `GrammarCompiler.compile_regex` is fine.

```suggestion
                self.ctx = compiler.compile_regex(self.config.regex_str)
```

Reply:
Sounds good. When I originally wrote this, the addition to `GrammarCompiler` wasn't there yet.